e2e/settings.spec.ts

```diff
@@ -10,6 +10,32 @@ test.describe("Authenticated settings Page", () => {
   test.beforeEach(async ({ page }) => {
     await loggedInAsUserOne(page);
   });
-  //
-  // Replace with tests for authenticated users
+  
+  // Test for changing username
+  test('Username input field', async ({ page }) => {
+    await page.goto('http://localhost:3000/settings', { timeout: 30000 });
+  
+    // Wait for the username input field to be visible
+    await page.locator('input[id="username"]').waitFor();
+  
+    // Test that the input field is visible and has the correct attributes
+    const inputField = page.locator('input[id="username"]');
+    await expect(inputField).toBeVisible();
+    await expect(inputField).toHaveAttribute('type', 'text');
+    await expect(inputField).toHaveAttribute('autocomplete', 'username');
+  
+    // Test that the error message appears when the input field is invalid
+    await inputField.fill('45&p^x#@!96%*()');
+    await page.locator('button[type="submit"]').click(); 
+    const errorMessage = page.locator('p:text-is("Username can only contain alphanumerics and dashes.")')
+    await expect(errorMessage).toBeVisible();
+    await expect(errorMessage).toHaveText('Username can only contain alphanumerics and dashes.');
+    // Reset the form 
+    await page.locator('button:has-text("Reset")').click();
+  
+    // Test that the input field can be filled with a valid value and saves it
+    await inputField.fill('codu-rules');
+    await page.locator('button[type="submit"]').click(); 
+    await expect(inputField).toHaveValue('codu-rules');
```

Comment:
Will this wait for the username change to take effect? 

If you fill in the new username and then click submit.

Will this test see the new username you have entered as the value of the input and complete straight away?

Reply:
I mean the tests pass for it. But it could be just catching the still filled in form from before the reset. 

Would you like me to change it to a page refresh instead? 

Or would you like some kind of timeOut in-between the state change and checking for the change to persist?

I would be happy to add either one 👍🏻

Reply:
Yeah I am not sure is there a loading indicator for these types of change? It would be nice to see

1. Click submit
2. Loading indicator appears
3. Loading indicator hidden
4. Check new username has taken effect